data/transactions/logic/eval.go

```diff
@@ -255,6 +258,9 @@ func (ep EvalParams) budget() int {
 	if ep.runModeFlags == runModeSignature {
 		return int(ep.Proto.LogicSigMaxCost)
 	}
+	if ep.PooledApplicationBudget > ep.Proto.MaxAppProgramCost {
+		return ep.PooledApplicationBudget
+	}
 	return ep.Proto.MaxAppProgramCost
```

Comment:
Not sure, but it looks like this is going to mean that a second app call is going to get 700, even if a first app call used 1399.  That is is, the first used up both budgets.